app/Http/Controllers/ContactController.php

```diff
@@ -2,42 +2,30 @@
 
 namespace App\Http\Controllers;
 
-use App\Http\Controllers\Controller;
-use App\Models\Contact;
-use Illuminate\Http\Request;
+use App\Contact;
+use App\Http\Requests\ContactRequest;
+use App\Services\ContactService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
 
 class ContactController extends Controller
 {
-    
+
     public function index()
     {
         //
     }
-    
-    public function store(ContactRequest $request)
+
+    public function store(ContactRequest $request): JsonResponse
     {
         $parameters = $request->validated();
 
-        $contacts = [];
-
-        foreach ($parameters['contacts'] as $keyContact => $contact) {
-
-            $columnsNames = [];
-
-            foreach ($parameters['columns'] as $keyColumn => $column) {
-                $columnsNames[] = $column['column-'.($keyColumn + 1)];
-            }
-
-            foreach ($columnsNames as $keyColumnName => $columnName) {
-                $contacts[$keyContact][$columnName] = $contact[$keyColumnName];
-            }
-        }
-
-        $contact = Contact::create($contacts);
+        $response = ContactService::store($parameters['contacts'], $parameters['columns']);
+        return response()->json($response, $response['success'] ? 200 : 400);
+    }
 
-        return response()->json([
-            'success' => true
-        ]);
+    public function getRecords(): JsonResponse
```

Comment:
**Additional feature**: A new `getRecords` method was added to provide data to a view that shows imported records.